engine/src/main/java/org/hibernate/search/engine/Version.java

```diff
@@ -22,7 +22,8 @@ private Version() {
 	 * @return A string representation of the version of Hibernate Search.
 	 */
 	public static String versionString() {
-		return Version.class.getPackage().getImplementationVersion();
+		// This implementation is replaced during the build with another one that returns the correct value.
+		return "UNKNOWN";
```

Comment:
Maybe we can use the API `Version.class.getPackage().getImplementationVersion()` to get the current version for the test. Because I think tests are never executed as natives...

Reply:
That wouldn't be useful for tests: `Version.class.getPackage().getImplementationVersion()` used not to work in tests, either. It was returning `null`.

I don't know why exactly, since it worked fine in real applications. Probably related to some classloading specifics of surefire/failsafe.

Anyway, it doesn't matter anymore, since that implementation is gone.